cpp/ql/lib/semmle/code/cpp/rangeanalysis/SimpleRangeAnalysis.qll

```diff
@@ -1549,7 +1549,8 @@ private float getGuardedUpperBound(VariableAccess guardedAccess) {
     // that there is one predecessor, albeit somewhat conservative.
     exists(unique(BasicBlock b | b = def.(BasicBlock).getAPredecessor())) and
     guardedAccess = def.getAUse(v) and
-    result = max(float ub | upperBoundFromGuard(guard, guardVa, ub, branch))
+    result = max(float ub | upperBoundFromGuard(guard, guardVa, ub, branch)) and
+    not exists(Expr e | e = guard.getAChild+() | convertedExprMightOverflow(e))
```

Comment:
Whenever we use transitive closures like this, we should double-check tuple counts and performance. But we might change the whole approach, so we don't need to do that now.

Reply:
Anders and I looked at the tuple counts together on the Asterisk project. The TC gets compiled to a `boundedFastTC` with a good restriction on the starting points. It produces 34M (compressed) rows in 7MB, which is not alarming, and DCA shows no measurable impact.